### PR TITLE
Fix sharding_pass_unittest stage3 precision problem

### PR DESCRIPTION
**Pull request by Wennie396**
### PR types

Bug fixes

### PR changes

Others

### Description

Pcard-70448
Fix sharding_pass_unittest stage3 precision problem.

### CODE REVIEW

File: test/auto_parallel/sharding_pass_unittest.py [hunk 1]
```diff
@@ -27,7 +27,7 @@
 def apply_pass(use_sharding=False, stage=None):
```
Comment (JZ-LIANG) [inline test/auto_parallel/sharding_pass_unittest.py:27]: Enable the unitest in cmake

File: test/auto_parallel/sharding_pass_unittest.py [hunk 2]
```diff
@@ -50,6 +50,7 @@ def apply_pass(use_sharding=False, stage=None):
 def reset_prog():
     paddle.fluid.framework.switch_main_program(paddle.static.Program())
     paddle.fluid.framework.switch_startup_program(paddle.static.Program())
+    paddle.utils.unique_name.switch()
 
 
 class TestShardingPass(unittest.TestCase):
```
Comment (JZ-LIANG) [inline test/auto_parallel/sharding_pass_unittest.py:56]: is the results exactly equal for runing dp twice?
if so, sharding should be exactly equal

File: test/auto_parallel/sharding_pass_unittest.py (hunk 1 above)
Comment (Wennie396) [reply test/auto_parallel/sharding_pass_unittest.py:27]: Done

File: test/auto_parallel/sharding_pass_unittest.py (hunk 2 above)
Comment (Wennie396) [reply test/auto_parallel/sharding_pass_unittest.py:56]: Done

**Review by JZ-LIANG — APPROVED** [review]
LTGM

**Pull request merged**